cmd/openshift-tuned.go

```diff
@@ -442,32 +462,55 @@ func mainLoop(clientset *kubernetes.Clientset, nodeName string) {
 			}
 
 		case err := <-wFs.Errors:
-			log.Printf("Error watching filesystem: %v", err)
-
-		case nodeEvent, ok := <-wNode.ResultChan():
-			if !ok {
-				log.Fatalf("Node event watch channel closed.")
-			}
-			nodeChangeHandler(nodeEvent, &tuned)
+			return fmt.Errorf("Error watching filesystem: %v", err)
 
 		case podEvent, ok := <-wPod.ResultChan():
 			if !ok {
-				log.Fatalf("Pod event watch channel closed.")
+				return fmt.Errorf("Pod event watch channel closed.")
 			}
 			podChangeHandler(podEvent, &tuned)
 
 		case <-tickerPull.C:
-			pullLabels(clientset, &tuned, nodeName)
+			if err := pullLabels(clientset, &tuned, nodeName); err != nil {
+				return err
+			}
 
 		case <-tickerReload.C:
-			timedTunedReloader(&tuned)
+			if err := timedTunedReloader(&tuned); err != nil {
+				return err
+			}
 		}
 	}
 }
 
-func main() {
-	var err error
+func supervise(clientset *kubernetes.Clientset, nodeName string) (err error) {
```

Comment:
This function name makes me feel some kinda way, mainLoop wasn't great but at least it was somewhat descriptive.